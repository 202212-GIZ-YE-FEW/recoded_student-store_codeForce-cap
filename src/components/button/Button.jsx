const BUTTON_STYLES = {
  orangeSignIn:
    "h-10 w-40 lg:w-48 lg:px-5 p-1 px-4 rounded-lg border-1 bg-pumpkin text-md font-normal text-white shadow-sm",
  forgotPassword:
    "h-10 w-40 lg:w-48 p-1 rounded-lg border-2 border-pumpkin bg-transparent font-normal text-md text-pumpkin shadow-sm",
  orangeSignUp:
    "h-10 w-36 mb-2 p-1 bg-pumpkin hover:border-2 hover:border-white font-normal text-white text-md rounded-lg",
  purpleSignIn:
    "h-9 w-36 m-1 p-1 rounded-lg border-2 bg-darkPurple font-light text-sm text-white shadow-sm",
  purpleSignUp:
    "h-10 w-40 m-1 p-1 rounded-lg border-2 bg-darkPurple font-light text-15px text-white shadow-sm",
  listItem:
    "h-12 w-72 md:h-14 md:w-80 lg:h-14 lg:w-96 m-1 p-1 rounded-lg border-2 bg-pumpkin font-md text-17px text-white shadow-sm",
  saveChanges:
    "h-12 w-56 lg:h-14 lg:w-64 m-1 p-1 px-2 hover:bg-purple hover:shadow-lg rounded-lg border-2 bg-pumpkin font-md text-17px text-white shadow-sm",
}

function Button({
  buttonStyle = "orangeSignIn",
  type = "button",
  text = "",
  handleClick,
}) {
  return (
    <button
      className={BUTTON_STYLES[buttonStyle]}
      onClick={handleClick}
      type={type}
    >
      {text}
    </button>
  )
}

export default Button
