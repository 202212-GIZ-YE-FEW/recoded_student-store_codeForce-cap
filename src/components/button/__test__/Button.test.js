import { fireEvent, render } from "@testing-library/react"
import renderer from "react-test-renderer"

import Button from "../Button"

describe("Button component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Button text='test' />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("calls handleClick function when button is clicked", () => {
    const handleClick = jest.fn()
    const { getByText } = render(
      <Button text='test' handleClick={handleClick} />
    )
    const button = getByText("test")
    fireEvent.click(button)
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it("applies the correct button style based on the buttonStyle prop", () => {
    const { getByText } = render(
      <Button text='test' buttonStyle='orangeSignUp' />
    )
    const button = getByText("test")
    expect(button.classList.contains("bg-pumpkin")).toBe(true)
    expect(button.classList.contains("rounded-lg")).toBe(true)
    expect(button.classList.contains("text-white")).toBe(true)
  })

  it("renders the correct button text based on the text prop", () => {
    const { getByText } = render(<Button text='test' />)
    const button = getByText("test")
    expect(button).toBeInTheDocument()
  })

  // it("renders the correct icon based on the icon prop", () => {
  //   const { getByTestId } = render(<Button text='test' icon='fa fa-plus' />)
  //   const icon = getByTestId("button-icon")
  //   expect(icon).toBeInTheDocument()
  // })

  it("renders with default props when no props are provided", () => {
    const { getByText } = render(<Button text='submit' />)
    const button = getByText("submit")
    expect(button.classList.contains("bg-pumpkin")).toBe(true)
    expect(button.classList.contains("rounded-lg")).toBe(true)
    expect(button.classList.contains("text-white")).toBe(true)
  })
})
