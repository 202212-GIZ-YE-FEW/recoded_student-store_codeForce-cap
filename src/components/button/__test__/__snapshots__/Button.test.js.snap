// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="h-10 w-40 lg:w-48 lg:px-5 p-1 px-4 rounded-lg border-1 bg-pumpkin text-md font-normal text-white shadow-sm"
  type="button"
/>
`;
